src/libgpu/src/glsl2/glsl2_tex.cpp

```diff
@@ -145,12 +145,18 @@ sampleFunc(State &state,
 
    auto resourceID = inst.word0.RESOURCE_ID();
    auto samplerID = inst.word2.SAMPLER_ID();
+   auto lodBias = inst.word1.LOD_BIAS();
 
    auto samplerDim = state.shader->samplerDim[samplerID];
-   auto samplerUsage = registerSamplerID(state, samplerID, isShadowOp);
-
-   if (resourceID != samplerID) {
-      throw translate_exception("Unsupported sample with RESOURCE_ID != SAMPLER_ID");
+   //Use resourceID to set the Usage
+   auto samplerUsage = registerSamplerID(state, resourceID, isShadowOp);
+
+   //The statement 'resourceID == samplerID' is true when the texture number is below 16.
+   //However, while sampler ID is limited to 16 max,max texture number allowed is 18. 
+   //So when use texture at index 16 and texture at index 17,the sampler id will be 0 and 1.
+   //When compile shader, the compiler will warn about this sampler overlap.
+   if (resourceID%16 != samplerID) {
+      throw translate_exception("Unsupported sample with RESOURCE_ID%16 != SAMPLER_ID");
```

Comment:
* `sample` -> `sampler` or perhaps
* change the message to something like "TextureID > 16. Sampler overlap will occur."